### DISCUSSION TITLE
realtime mesh interpolation with modes like in A animate or blender, instead of creating it manualy into level strip

### DISCUSSION THREAD

**Discussion by Q210108** [opentoonz/opentoonz#6498] [Ideas]
I suggest this because, create interpolation between 2 frames into level strip is less flexible than automaticaly calculate it las we do it for move, resize, rotate with keys . That allow us to control interpolation speed and save many time

**Comment by RodneyBaker** [comment]
You will likely need to provide some screenshots, video, links or other references to flesh out this request.
Otherwise you may not get the feature you expect.

**Comment by artisteacher** [comment]
@Q210108 have you tried the plastic tool yet?

**Comment by Q210108** [comment]
> You will likely need to provide some screenshots, video, links or other references to flesh out this request. Otherwise you may not get the feature you expect.

https://user-images.githubusercontent.com/100626161/158524369-92b245db-8596-484d-968f-5dbd0e5812e3.mp4

For example in adobe animate, there is movement interpolation based on vector like in level strip, but in animate,we have more flexibility, because it automaticaly calculate frames between 2 frames,and we can move frames because it's like move animate tool or rotate animate tool in opentoonz but with vector movement animate tool

**Comment by RodneyBaker** [comment]
Ah... the oft desired control point animation (or curve animation as some refer to it).
When you used the term 'realtime mesh interpolation' that sent us in a different direction.

There was an attempt some years back to add that capability but the developer decided the effort needed to be restarted from scratch.

I'm confident we'll get there eventually but because that level of animation deals with one of the most basic objects Opentoonz uses (control points)  it will take time to implement.  There are also some competing interests in the same general area of interest with regard to SVG support as well as some user desires to explore other vector based approaches.  

The title of this requests seems to place a focus on 'meshes' which is an abstraction away from the control point level of manipulation.  As @artisteacher mentioned, currently the Plastic Tool is the way to achieve that movement in Opentoonz and likely will be until we gain control point level animation.

There are a few request open for control point animation so unless this report can add something to those we'll probably close this as a dupliate request.  I'll plan to reference this report in those reports so the larger discussion and level of interest is easier to follow.
